File: backend/Model/admission.py
from sqlalchemy import Column, Integer, String, ForeignKey, Date, Boolean, DateTime
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from config.db import Base

class Admission(Base):
    __tablename__ = 'admissions'

    id = Column(Integer, primary_key=True, nullable=False)
    student_id = Column(Integer, ForeignKey('students.id'), nullable=False)
    admission_date = Column(Date, nullable=False)
    previous_school = Column(String(255), nullable=True)
    guardian_name = Column(String(255), nullable=False)
    guardian_number = Column(String(20), nullable=False)
    isActive = Column(Boolean, default=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

    student = relationship(
        "Student",
        back_populates="admissions"
    )

File: backend-express/src/models/Admission.js
import mongoose from 'mongoose';

const admissionSchema = new mongoose.Schema(
  {
    student_id: { type: mongoose.Schema.Types.ObjectId, ref: 'Student', required: true },
    admission_date: { type: Date, required: true },
    previous_school: { type: String },
    guardian_name: { type: String, required: true, trim: true },
    guardian_number: { type: String, required: true, trim: true },
    isActive: { type: Boolean, default: true },
  },
  { timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' } }
);

admissionSchema.virtual('student', {
  ref: 'Student',
  localField: 'student_id',
  foreignField: '_id',
  justOne: true,
});

admissionSchema.index({ isActive: 1 });
admissionSchema.index({ student_id: 1 });

export const Admission = mongoose.model('Admission', admissionSchema);
